src/raster/array.jl

```diff
@@ -0,0 +1,38 @@
+Base.size(dataset::Dataset) = begin
+    band = getband(dataset, 1)
+    width(band), height(band), nraster(dataset)
+end
+
+const AllowedXY = Union{Integer,Colon,AbstractRange}
+const AllowedBand = Union{Integer,Colon,AbstractArray}
+
+Base.getindex(dataset::Dataset, x::AllowedXY, y::AllowedXY, bands::AllowedBand) = begin
```

Comment:
Can we keep it `bands::Union{Integer,Colon,AbstractArray}`? Do we foresee `AllowedBand` being used more broadly in the codebase?

Reply:
similarly, inlining `AllowedXY` and removing the const definition.

Reply:
I was making it very explicit that band and x/y actually have different allowed input. It will be used again for `setindex!` when we write that.